lib/fleiss/worker.rb

```diff
@@ -86,10 +84,11 @@ def perform(job)
       finished ? job.finish(owner) : job.reschedule(owner)
     end
   rescue StandardError => e
-    handle_exception e, "processing job ##{job.id} (by thread #{thread_id})"
+    log_exception e, "processing job ##{job.id} (by thread #{thread_id})"
+    raise
```

Comment:
Do not suppress error here, bubble it up to `Fleiss.backend.around_perform` for backend-specific handling.

In the end, error will be suppressed in `run_cycle` ((almost) top-level)

Reply:
So I'm not sure about logging the same error 2x, it doesn't make much sense TBH. All errors should be handled here, there's no reason to bubble them up. The rescue in run_cycle is a last resort safety net but it should never actually trigger

Reply:
@dim so...

We have this:

```ruby
Fleiss.backend.wrap_perform do
  perform(...) # I wanted error not to be suppressed in this method, but be raised so it can be handled in wrap_perform
end
```

We do rescue from StatementInvalid in the backend's around_perform method.

I think I just need to re-org it like this:

```
def perform
  Fleiss.backend.wrap_perform do
    ...
  end
rescue StandardError => err
  # log, but not re-raise
end
```

Will push in a min...